openlane/scripts/magic/lef.tcl

```diff
@@ -45,6 +45,6 @@ if { $::env(MAGIC_WRITE_FULL_LEF) } {
     lef write $::env(STEP_DIR)/$::env(DESIGN_NAME).lef
 } else {
     puts "\[INFO] Writing abstract LEF"
-    lef write $::env(STEP_DIR)/$::env(DESIGN_NAME).lef -hide
+    lef write $::env(STEP_DIR)/$::env(DESIGN_NAME).lef -hide -pinonly
```

Comment:
Sorry, I can't access opencircuitdesign.com right now-- what does `-pinonly` do? We might wanna put it behind a flag depending on what it does…

Reply:
Without it, In the LEF instead of listing just the area defined as "pin" in the ports, it exports the whole metal area connected to it. 

See for instance :
![](https://i.imgur.com/cUUaUyH.png)

You can see that what got exported as the pin area is actually all the `met4` parts of that net and not just the end.

Reply:
Makes sense. @kareefardi Thoughts?

Reply:
I would wrap it around a variable.

Reply:
Should the default be the previous behavior or the new behavior then?

Reply:
Resolution: Adding a variable